Sources/Apollo/LegacyCacheReadInterceptor.swift

```diff
@@ -27,6 +30,35 @@ public class LegacyCacheReadInterceptor: ApolloInterceptor {
                          completion: completion)
     case .query:
       switch request.cachePolicy {
+      case .fetchReturningCacheDataOnError:
+        chain.proceedAsync(
+          request: request,
+          response: response,
+          completion: { fetchResult in
+            // if the fetch request failed, try to read from the cache
+            switch fetchResult {
+            case let .success(fetchData):
+              chain.returnValueAsync(for: request, value: fetchData, completion: completion)
+
+            case let .failure(fetchError):
+              self.fetchFromCache(for: request, chain: chain) { cacheResult in
```

Comment:
What types of failures are returned in this case?

I would assume we would only want to return cache data when a network connection occurred.

Reply:
Yes, primarily this will be network errors, but it might also be JSONDecoding errors (such as if the server is misbehaving in its responses).

In general using this policy will either:
1. return a success with the data you expected (either from the server or cache)
1. return a `CacheError` that provides the `networkError` (again, likely network stuff but might also be parsing errors) and a `cacheError` that provides why the cache data was unavailable (most likely `.missingValue`)